src/megaclient.cpp

```diff
@@ -2841,13 +2841,7 @@ int MegaClient::preparewait()
         nexttransferretry(GET, &nds);
 
         // retry transferslots
-        for (transferslot_list::iterator it = tslots.begin(); it != tslots.end(); it++)
-        {
-            if (!(*it)->retrybt.armed())
-            {
-                (*it)->retrybt.update(&nds);
-            }
-        }
+        tslotsbackoff.update(&nds, false);
```

Comment:
Don't we do the opposite than before in here?
It used to update the timer when the backoff was not armed, but the `BackoffTimerGroupTracker::update()`, for slots, does update for those which are armed.

Reply:
I think that check for !armed might be an attempt at efficiency, or a way to avoid that next=1 but not processed due to the `xferpaused` flag issue discussed above.   The thing is, the profiler showed this loop took a lot of CPU when we push the number of active transfers up to say, 1024, as the entire list was iterated on every wait.   So now, instead of checking every tslot, we specifically only keep a list of those tslots that are in a backoff state - so this loop only costs us when something goes wrong, and not in normal operation.   Also the items in the list correspond to that test, they will be in the list if `!armed()`  (with some adjustments  -eg. NEVER items are not in the list either, as we know those won't set the `nds`.   Also, there could be some whose `next` is set, and the Waiter::ds has ticked over so the transfer should now be checked - and we should call update() on those so `nds` goes to 0, but in fact the old loop didn't do that)